src/invidious/helpers/serialized_yt_data.cr

```diff
@@ -274,4 +274,63 @@ struct Continuation
   end
 end
 
+struct HashtagPage
+  include DB::Serializable
+
+  property videos : Array(SearchItem) | Array(Video)
+  property header : HashtagHeader?
+
+  def to_json(locale : String?, json : JSON::Builder)
+    json.object do
+      json.field "type", "hashtag"
+      if self.header != nil
+        json.field "header" do
+          self.header.to_json(json)
+        end
+      end
+      json.field "results" do
+        json.array do
+          self.videos.each do |item|
+            item.to_json(locale, json)
+          end
+        end
+      end
+    end
+  end
+
+  # TODO: remove the locale and follow the crystal convention
+  def to_json(locale : String?, _json : Nil)
+    JSON.build do |json|
+      to_json(locale, json)
+    end
+  end
+
+  def to_json(json : JSON::Builder)
+    to_json(nil, json)
+  end
+end
+
+struct HashtagHeader
+  include DB::Serializable
+
+  property tag : String
+  property channel_count : Int64
+  property video_count : Int64
+
+  def to_json(json : JSON::Builder)
+    json.object do
+      json.field "type", "hashtagHeader"
+      json.field "hashtag", self.tag
+      json.field "channelCount", self.channel_count
+      json.field "videoCount", self.video_count
+    end
+  end
+
+  def to_json(_json : Nil)
+    JSON.build do |json|
+      to_json(json)
+    end
+  end
+end
```

Comment:
Maybe we could use a generic search metadata struct here? The idea is to reuse it for regular search (we'd add `estimatedResults`).

Reply:
I'm not sure i understand the changes you'd like here. Do i just add estimatedResults and change the struct name?